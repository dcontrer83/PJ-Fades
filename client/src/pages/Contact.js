import React, {useState} from 'react';

import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';

const Contact = () => {

    const [formData, setFormData] = useState({
        fullname: '',
        email: '',
        message: '',
    })

    const style = {
        offset: {
            height: "93px"
        },
        main: {
            height: "100%"
        },
        misc: {
            marginTop: "40px",
            marginBottom: "50px"
        },
        sendTitle: {
            padding: "25px"
        },
        formBox: {
            height: "500px",
        },
        formBox2: {
            width: "60%"
        }
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        //send it to database or send it through email?
    }

    const handleInputChange = (e) => {
        const {name, value} = e.target;

        setFormData({
            ...formData,
            [name]: value,
        })
    }
    return (
        <main>
            <div style={style.offset}></div>
            <div style={style.main}>
                <div className='text-center' style={style.misc}>
                    <h1>Contact</h1>
                    <p className='fs-3'>Phone number: 1-951-553-4409</p>
                    <p className='fs-3'>DM me on <a href="https://instagram.com/pushnpj?igshid=YmMyMTA2M2Y=">Instagram!</a></p>
                </div>
                <div className='bg-dark text-white mx-auto' style={style.formBox}>
                    <h2 className='text-center mb-0' style={style.sendTitle}>Send a Message!</h2>
                    <Form data-testid="form-test" className='mx-auto' style={style.formBox2} onSubmit={handleSubmit}>
                        <Form.Group className='mb-3' controlId='contact-form-name'>
                            <Form.Label className='mt-3'>Full Name:</Form.Label>
                            <Form.Control data-testid="input-name"
                            type="text" 
                            name="fullname"
                            value={formData.fullname}
                            onChange={handleInputChange}
                            placeholder="name"
                            ></Form.Control>
                        </Form.Group>
                        <Form.Group className='mb-3' controlId='contact-form-email'>
                            <Form.Label>Email:</Form.Label>
                            <Form.Control data-testid="input-email"
                            type="email" 
                            name="email"
                            value={formData.email}
                            onChange={handleInputChange}
                            placeholder='Email address'
                            ></Form.Control>
                        </Form.Group>
                        <Form.Group className='mb-3' controlId='contact-form-message'>
                            <Form.Label>Message:</Form.Label>
                            <Form.Control data-testid="input-message"
                            as="textarea" 
                            rows={3}
                            name="message"
                            value={formData.message}
                            onChange={handleInputChange}
                            placeholder="message"
                            ></Form.Control>
                        </Form.Group>
                        <Button data-testid="submit-btn" type="submit">
                            Submit
                        </Button>
                    </Form>
                </div>
            </div>
        </main>
    )
}

export default Contact;