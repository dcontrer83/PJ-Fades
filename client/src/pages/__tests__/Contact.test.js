import React from 'react';
import { createRoot } from 'react-dom/client';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import Contact from '../Contact';
import '@testing-library/jest-dom';
import { render, screen, fireEvent, waitFor, getByTestId } from '@testing-library/react';

globalThis.IS_REACT_ACT_ENVIRONMENT = true;

let root;
let container;

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    act(() => {
        root = createRoot(container);
    })
});

afterEach(() => {
    act(() => {
        root.unmount();
    })
    container.remove();
    container = null;
});

describe('Form', () => {
    it('should change values' , () => {
        render(<Contact />);
        
        fireEvent.change(screen.getByTestId('input-name'), { target: { value: 'John Doe'}});
        fireEvent.change(screen.getByTestId('input-email'), { target: { value: 'someUser@email.com'}});
        fireEvent.change(screen.getByTestId('input-message'), { target: { value: 'Hello user'}});

        expect(screen.getByTestId('input-name').value).toEqual('John Doe');
        expect(screen.getByTestId('input-email').value).toEqual('someUser@email.com');
        expect(screen.getByTestId('input-message').value).toEqual('Hello user');

    });
});

let div1Element;
let div2Element;
let div3Element;
let div4Element;

function getAllDivElements() {
    render(<Contact />);
    div1Element = screen.getByTestId('div-1-element');
    div2Element = screen.getByTestId('div-2-element');
    div3Element = screen.getByTestId('div-3-element');
    div4Element = screen.getByTestId('div-4-element');
}

describe('Div elements', () => {
    it('should be in the document', () => {
        getAllDivElements();
        expect(div1Element).toBeInTheDocument();
        expect(div2Element).toBeInTheDocument();
        expect(div3Element).toBeInTheDocument();
        expect(div4Element).toBeInTheDocument();
    });
    it('should have specific style', () => {
        getAllDivElements();
        expect(div1Element).toHaveStyle('height: 93px');
        expect(div2Element).toHaveStyle('height: 100%');
        expect(div3Element).toHaveStyle('marginTop: 40px; marginBottom: 50px;');
        expect(div4Element).toHaveStyle('height: 500px');
    });
    it('should have specific className', () => {
        getAllDivElements();
        expect(div3Element).toHaveClass('text-center');
        expect(div4Element).toHaveClass('bg-dark text-white mx-auto');
    });
});

let h1Element;
let h2Element;

function getAllHeadingElements() {
    render(<Contact />);
    h1Element = screen.getByTestId('heading-1-element');
    h2Element = screen.getByTestId('heading-2-element');
}

describe('Heading elements', () => {
    it('should be in the document', () => {
        getAllHeadingElements();
        expect(h1Element).toBeInTheDocument();
        expect(h2Element).toBeInTheDocument();
    });
    it('should have specific text content', () => {
        getAllHeadingElements();
        expect(h1Element).toHaveTextContent('Contact');
        expect(h2Element).toHaveTextContent('Send a Message!');
    });
});

let p1Element;
let p2Element;
let p2AElement;

function getAllParagraphElements() {
    render(<Contact />);
    p1Element = screen.getByTestId('p-1-element');
    p2Element = screen.getByTestId('p-2-element');
    p2AElement = screen.getByTestId('p-2-a-element');
}

describe('Paragraph elements', () => {
    it('should be in the document', () => {
        getAllParagraphElements();
        expect(p1Element).toBeInTheDocument();
        expect(p2Element).toBeInTheDocument();
        expect(p2AElement).toBeInTheDocument();
    });
    it('should have specific className', () => {
        getAllParagraphElements();
        expect(p1Element).toHaveClass('fs-3');
        expect(p2Element).toHaveClass('fs-3');
    });
    it('should have specific text content', () => {
        getAllParagraphElements();
        expect(p1Element).toHaveTextContent('Phone number: 1-951-553-4409');
        expect(p2Element).toHaveTextContent('DM me on Instagram!');
        expect(p2AElement).toHaveTextContent('Instagram!');
    });
    it('should redirect to proper link', () => {
        getAllParagraphElements();
        expect(p2AElement.href).toBe('https://instagram.com/pushnpj?igshid=YmMyMTA2M2Y=');
    });
});

let formElement;

function getFormElement() {
    render(<Contact />);
    formElement = screen.getByTestId('form-element');
}

describe('Form element', () => {
    it('should be in the document', () => {
        getFormElement();
        expect(formElement).toBeInTheDocument();
    });
    it('should have specific className', () => {
        getFormElement();
        expect(formElement).toHaveClass('mx-auto');
    });
    it('should have specific style', () => {
        getFormElement();
        expect(formElement).toHaveStyle('width: 60%');
    });
});
