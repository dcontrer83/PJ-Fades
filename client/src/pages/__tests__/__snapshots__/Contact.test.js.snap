// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should match snapshot 1`] = `
"<main>
  <div data-testid=\\"div-1-element\\" style=\\"height: 93px;\\"></div>
  <div data-testid=\\"div-2-element\\" style=\\"height: 100%;\\">
    <div data-testid=\\"div-3-element\\" class=\\"text-center\\" style=\\"margin-top: 40px; margin-bottom: 50px;\\">
      <h1 data-testid=\\"heading-1-element\\">Contact</h1>
      <p data-testid=\\"p-1-element\\" class=\\"fs-3\\">Phone number: 1-951-553-4409</p>
      <p data-testid=\\"p-2-element\\" class=\\"fs-3\\">DM me on <a data-testid=\\"p-2-a-element\\" href=\\"https://instagram.com/pushnpj?igshid=YmMyMTA2M2Y=\\">Instagram!</a></p>
    </div>
    <div data-testid=\\"div-4-element\\" class=\\"bg-dark text-white mx-auto\\" style=\\"height: 500px;\\">
      <h2 data-testid=\\"heading-2-element\\" class=\\"text-center mb-0\\" style=\\"padding: 25px;\\">Send a Message!</h2>
      <form data-testid=\\"form-element\\" style=\\"width: 60%;\\" class=\\"mx-auto\\">
        <div data-testid=\\"form-group-element\\" class=\\"mb-3\\"><label class=\\"mt-3 form-label\\" for=\\"contact-form-name\\" data-testid=\\"label-name-element\\">Full Name:</label><input data-testid=\\"input-name\\" name=\\"fullname\\" placeholder=\\"name\\" type=\\"text\\" id=\\"contact-form-name\\" class=\\"form-control\\" value=\\"\\"></div>
        <div class=\\"mb-3\\"><label class=\\"form-label\\" for=\\"contact-form-email\\" data-testid=\\"label-email-element\\">Email:</label><input data-testid=\\"input-email\\" name=\\"email\\" placeholder=\\"Email address\\" type=\\"email\\" id=\\"contact-form-email\\" class=\\"form-control\\" value=\\"\\"></div>
        <div class=\\"mb-3\\"><label class=\\"form-label\\" for=\\"contact-form-message\\" data-testid=\\"label-message-element\\">Message:</label><textarea data-testid=\\"input-message\\" rows=\\"3\\" name=\\"message\\" placeholder=\\"message\\" id=\\"contact-form-message\\" class=\\"form-control\\"></textarea></div><button type=\\"submit\\" data-testid=\\"submit-btn\\" class=\\"btn btn-primary\\">Submit</button>
      </form>
    </div>
  </div>
</main>"
`;
